### PR TITLE
new_audit: td-has-header

### PR DESCRIPTION
**Pull request by jazyan**
See http://go/prcpg for the current list of audits to be enabled.

Surprisingly, `a11y_tester.html` already has [a specific test case](https://github.com/GoogleChrome/lighthouse/blob/main/cli/test/fixtures/a11y/a11y_tester.html#L209-L217) for this.

### CODE REVIEW

**Comment by connorjclark** [comment]
This audit has some funny history. #6169

**Comment by connorjclark** [comment]
Can you fill out the expected audit result for `td-has-header` in `cli/test/smokehouse/test-definitions/a11y.js`? Ran via `yarn smoke a11y`

**Comment by paulirish** [comment]
fwiw This is the test failure I'm seeing in the CI logs: 
image

**Comment by jazyan** [comment]
Added `td-has-header` case to a11y smoke tests, and updated the category renderer unit test. 

Looking into the DevTools e2e failing assertions 👀 

(and whoops, accidentally re-added commits after fetch from main.)

File: core/audits/accessibility/td-has-header.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * @fileoverview Ensure that large tables have `[header]` attributes.
+ * See base class in axe-audit.js for audit() implementation.
+ */
+
+import AxeAudit from './axe-audit.js';
+import * as i18n from '../../lib/i18n/i18n.js';
+
+const UIStrings = {
+  /** Title of an accesibility audit that evaluates if all large table elements use the headers HTML attribute. This title is descriptive of the successful state and is shown to users when no user action is required. */
+  title: '`<td>` elements in a large `<table>` has one or more table headers.',
```
Comment (brendankenny) [inline core/audits/accessibility/td-has-header.js:17]:
```suggestion
  title: '`<td>` elements in a large `<table>` have one or more table headers.',
```

File: core/audits/accessibility/td-has-header.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  /** Title of an accesibility audit that evaluates if all large table elements use the headers HTML attribute. This title is descriptive of the failing state and is shown to users when there is a failure that needs to be addressed. */
+  failureTitle: '`<td>` elements in a large `<table>` do not have one or more table headers.',
```
Comment (brendankenny) [inline core/audits/accessibility/td-has-header.js:19]: maybe 
```suggestion
  failureTitle: '`<td>` elements in a large `<table>` do not have table headers.',
```

File: core/audits/accessibility/td-has-header.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  /** Description of a Lighthouse audit that tells the user *why* they should try to pass. This is displayed after a user expands the section to see more. No character length limits. The last sentence starting with 'Learn' becomes link text to additional documentation. */
+  description: 'Screen readers have features to make navigating tables easier. Ensuring ' +
+      'that `<td>` elements in a large (3 or more cells wide and high) table have an associated ' +
```
Comment (brendankenny) [inline core/audits/accessibility/td-has-header.js:22]: maybe
```suggestion
      'that `<td>` elements in a large (3 or more cells in width and height) table have an associated ' +
```
? Not sure if it's better but the first time I read the current version I had to re-read to get what the parenthetical was saying

**Review by brendankenny — APPROVED** [review]
just some potential `UIStrings` suggestions but otherwise LGTM!

File: core/audits/accessibility/td-has-header.js (hunk 3 above)
Comment (brendankenny) [reply core/audits/accessibility/td-has-header.js:22]: and maybe even
```suggestion
      'that `<td>` elements in a large table (3 or more cells in width and height) have an associated ' +
```
to keep the adjective with the noun, but really getting into the nits now

**Pull request merged**